perun-core/src/main/java/cz/metacentrum/perun/core/impl/SearcherImpl.java

```diff
@@ -186,62 +183,37 @@ private void insertWhereClausesAndQueryParametersFromAttributes(StringBuilder qu
 			query.append("left join attr_names nam").append(counter).append(" on val").append(counter).append(".attr_id=nam").append(counter).append(".id ");
 
 			if (value == null || value.isEmpty()) {
-				if(key.getType().equals(LinkedHashMap.class.getName()) ||
-						key.getType().equals(BeansUtils.largeStringClassName) ||
-						key.getType().equals(BeansUtils.largeArrayListClassName)) {
-					whereClauses.add("val" + counter + ".attr_value_text IS NULL ");
-				} else {
-					whereClauses.add("val" + counter + ".attr_value IS NULL ");
-				}
+				whereClauses.add("val" + counter + ".attr_value IS NULL ");
 			} else {
 				if (key.getType().equals(Integer.class.getName())) {
 					key.setValue(Integer.valueOf(value));
 					whereClauses.add("val" + counter + ".attr_value=:v" + counter + " ");
 					whereClauses.add("nam" + counter + ".type=:n" + counter + " ");
 					parameters.addValue("n" + counter, Integer.class.getName());
 					parameters.addValue("v" + counter, BeansUtils.attributeValueToString(key));
-				} else if (key.getType().equals(String.class.getName())) {
+				} else if (key.getType().equals(String.class.getName()) || key.getType().equals(BeansUtils.largeStringClassName)) {
```

Comment:
There is nothing like large objects anymore, so can't we remove these special parts too?

Reply:
I'd like to, but for now we have existing attributes with these types and I'm not sure, that nobody has that harcoded when calling API. But If we agree and let others know, we can convert all such attributes to their default type using DB changelog and then it can be removed from everywhere (gui/cli/core).